app/Cargo.toml

```diff
@@ -30,7 +30,9 @@ log = { workspace = true }
 tokio = { version = "1", features = ["rt-multi-thread"] }
 futures = "0.3"
 async-std = "1"
-tower-lsp = { version = "0.17", default-features = false, features = ["runtime-agnostic"] }
+tower-lsp = { git = "https://github.com/tower-lsp/tower-lsp", rev = "19f5a87810ff4b643d2bc394e438450bd9c74365", default-features = false, features = [
```

Comment:
I think we can deduplicate this dependency and the declaration in `lang/lsp/Cargo.toml` by specifying it in the workspace `Cargo.toml`.